slow_odgi/examples.md

```diff
@@ -0,0 +1,157 @@
+# Pollen language walkthrough
+
+## Core types in Pollen
+
+    type Graph = {
+      segments: Set<Segment>;
+      paths: Set<Path>;
+      links: Set<Links>; 
+    }
+
+    type Segment = {
+      sequence: Strand; //ACTGAC, etc. 
+      links: Set<Link>; //links encode their direction + orientation
+      steps: Set<Step>; //steps that go through segment
+    }
+
+    type Handle = {
+      segment: Segment; 
+      orientation: bool;
+    }
+
+    type Path = {
+      steps: List<Step>;
+    }
+
+    type Step = {
+      path: Path; 
+      idx: int; //where in the path is this sequence?
+      handle: Handle; //segment + orientation
+    }
+
+    type Base = A | C | T | G | N
+    type Strand = list Base
+
+    type Link = {
+      start: Handle; 
+      end: Handle;
+    }
+
+## Chop
+
+## Crush
+
+    graph out_g;
+    parset out_segs[Segment, out_g];
+
+    for segment in Segments {
+      seq = Strand();
+      in_n = false;
+      for c in segment.sequence {
+        if c == 'N' {
+          if !in_n {
+            in_n = true;
+            seq.push(c);
+          }
+        } else {
+            in_n = false;
+            seq.push(c);
+        }
+      }
+      emit { segment with sequence: seq } to out_segs;
+    }
+
+## Degree
+
+    // Output is a (Segment, int) array.
+    outset out_degs[Segment, int];
```

Comment:
Further questions about what the stuff in the square brackets mean. Does this mean that the output set contains `(Segment, int)` tuples? If so, it seems like the more obvious syntax would be `out_degs[(Segment, int)]`. Otherwise, something like `out_stuff[Segment]` would seem to contain 1-element tuples, which doesn't seem right.

Reply:
Yeah, definitely wanted to rework that because it felt clunky. Thanks for the suggestion!